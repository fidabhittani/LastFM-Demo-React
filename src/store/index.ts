/**
 * Application redux store with configured middlewares and reducers.
 */

import { applyMiddleware, createStore } from "redux";
import { composeWithDevTools } from "redux-devtools-extension";
import reducers from "reducers";
import thunkMiddleware from "redux-thunk";

export const store = createStore(
  reducers,
  composeWithDevTools(applyMiddleware(thunkMiddleware))
);

export const dispatch = store.dispatch;

export default store;
