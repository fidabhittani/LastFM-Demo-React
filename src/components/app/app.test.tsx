import React from "react";
import ReactDOM from "react-dom";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { shallow } from "enzyme";

import Root from "./index";
import store from "store";

/** ROOT TEST */
describe("App Root Component Tests", () => {
  const WrappedReduxRoot = () => (
    <BrowserRouter>
      <Provider store={store}>
        <Root key={"store"} />
      </Provider>
    </BrowserRouter>
  );

  it("Root Component renders correctly", () => {
    const root = shallow(<WrappedReduxRoot />);
    expect(root).toMatchSnapshot();

    const div = document.createElement("div");
    ReactDOM.render(<WrappedReduxRoot />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
});
