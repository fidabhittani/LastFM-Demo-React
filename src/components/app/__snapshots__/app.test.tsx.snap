// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Root Component Tests Root Component renders correctly 1`] = `ShallowWrapper {}`;
